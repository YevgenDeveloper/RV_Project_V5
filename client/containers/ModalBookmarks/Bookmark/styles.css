@import 'client/styles/variables';

.bookmark {
  display: flex;
  flex-flow: row nowrap;
  margin-bottom: 20px;
  width: 100%;
  max-width: 400px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  height: 60px;
  text-align: left;
  font-size: $font-size-default;

  &:hover {
    cursor: pointer;
  }
}

.image {
  display: block;
  flex-shrink: 0;
  width: auto;
  height: 50px;
  margin-right: 15px;
}

.info {
  width: 100%;
  max-width: 400px;
  margin-bottom: 3px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  user-select: text;
  * {
    user-select: text;
  }
}
