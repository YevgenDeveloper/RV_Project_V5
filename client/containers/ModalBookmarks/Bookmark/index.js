// @flow

import type { BookmarkType } from 'client/types';

import React from 'react';

import { googleStorageUrl } from 'client/constants';
import styles from './styles.css';

type Props = BookmarkType & {
  onClick: void => void,
};

const Bookmark = (props: Props) => {
  const title = `Ride: ${props.ride}, camera: ${props.camera}`;

  return (
    <button
      type="button"
      className={styles.bookmark}
      onClick={() => props.onClick()}
    >
      <img
        alt={title}
        className={styles.image}
        src={`${googleStorageUrl}/${props.ride}/${props.camera}/${props.image}`}
      />
      <div>
        <div className={styles.info}>
          <b>Ride:</b>
          {` ${props.ride}`}
        </div>
        <div className={styles.info}>
          <b>Camera:</b>
          {` ${props.camera}`}
        </div>
      </div>
    </button>
  );
};

export default Bookmark;
