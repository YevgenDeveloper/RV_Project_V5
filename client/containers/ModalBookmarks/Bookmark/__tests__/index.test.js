// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import Bookmark from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(
    <Bookmark ride="Test" camera="Test" image="Test" onClick={jest.fn()} />,
  );

  expect(wrapper).toMatchSnapshot();
});
