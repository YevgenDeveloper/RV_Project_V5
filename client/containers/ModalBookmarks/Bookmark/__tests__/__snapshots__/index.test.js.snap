// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bookmark
    camera="Test"
    image="Test"
    onClick={[MockFunction]}
    ride="Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Ride: Test, camera: Test"
          className="image"
          src="https://storage.googleapis.com/railview/Test/Test/Test"
        />,
        <div>
          <div
            className="info"
          >
            <b>
              Ride:
            </b>
             Test
          </div>
          <div
            className="info"
          >
            <b>
              Camera:
            </b>
             Test
          </div>
        </div>,
      ],
      "className": "bookmark",
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Ride: Test, camera: Test",
          "className": "image",
          "src": "https://storage.googleapis.com/railview/Test/Test/Test",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="info"
            >
              <b>
                Ride:
              </b>
               Test
            </div>,
            <div
              className="info"
            >
              <b>
                Camera:
              </b>
               Test
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Ride:
                </b>,
                " Test",
              ],
              "className": "info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ride:",
                },
                "ref": null,
                "rendered": "Ride:",
                "type": "b",
              },
              " Test",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Camera:
                </b>,
                " Test",
              ],
              "className": "info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Camera:",
                },
                "ref": null,
                "rendered": "Camera:",
                "type": "b",
              },
              " Test",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Ride: Test, camera: Test"
            className="image"
            src="https://storage.googleapis.com/railview/Test/Test/Test"
          />,
          <div>
            <div
              className="info"
            >
              <b>
                Ride:
              </b>
               Test
            </div>
            <div
              className="info"
            >
              <b>
                Camera:
              </b>
               Test
            </div>
          </div>,
        ],
        "className": "bookmark",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Ride: Test, camera: Test",
            "className": "image",
            "src": "https://storage.googleapis.com/railview/Test/Test/Test",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="info"
              >
                <b>
                  Ride:
                </b>
                 Test
              </div>,
              <div
                className="info"
              >
                <b>
                  Camera:
                </b>
                 Test
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Ride:
                  </b>,
                  " Test",
                ],
                "className": "info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Ride:",
                  },
                  "ref": null,
                  "rendered": "Ride:",
                  "type": "b",
                },
                " Test",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Camera:
                  </b>,
                  " Test",
                ],
                "className": "info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Camera:",
                  },
                  "ref": null,
                  "rendered": "Camera:",
                  "type": "b",
                },
                " Test",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
